jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java

```diff
@@ -346,6 +346,14 @@ protected ServerSocketChannel openAcceptChannel() throws IOException
             }
             catch (BindException e)
             {
+                try
+                {
+                    serverChannel.close();
+                }
+                catch (IOException ioe)
+                {
+                    LOG.warn(ioe);
+                }
```

Comment:
Can you please rework this fix to use try-with-resources, so there is no need of an explicit close?

Reply:
We only want the serverChannel closed if the BindException occurs.  Otherwise it needs to be returned to the caller in an open state.  I don't believe try-with-resources is what we want here but let me know if I misunderstood.

Reply:
@sbordet try-with-resources doesn't work in this case as we use the socket in the successful case...  actually probably should null out the field... just in case....

Reply:
Actually no fields were set, but there was one already in a try-with-resources, so backed the change out of that one.

Reply:
Looks like the serverChannel is within the method scope so nothing to null out. I looked at the caller scope or otherwise to see if there was a better place to adjust the close (such as a try with resources).  And I'm not seeing that.  For the specific issue observed the change as I have it closes that problem.  If this is part of a larger concern I'm not sure I follow.  Do you need any change from the patch as-is?

Reply:
Re: using try-with-resources: doh :)

@gregw #5802 is the same fix? @joewitt can you use `IO.close()` instead? Also, I think we want to catch everything that may possibly be throws after `ServerSocketChannel.open()`, so I would widen the `try` scope.

Reply:
Incorporating this change within #5802.